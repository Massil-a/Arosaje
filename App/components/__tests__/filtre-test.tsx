import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import ActuFiltre from '../../app/actunav/actufiltre';
import AsyncStorage from '@react-native-async-storage/async-storage';

jest.mock('@react-native-async-storage/async-storage', () => ({
    setItem: jest.fn(),
    getItem: jest.fn(),
    removeItem: jest.fn(),
    clear: jest.fn(),
}));

describe('ActuFiltre', () => {
  it('renders correctly', () => {
    const { toJSON } = render(
      <NavigationContainer>
        <ActuFiltre navigation={{}} />
      </NavigationContainer>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
