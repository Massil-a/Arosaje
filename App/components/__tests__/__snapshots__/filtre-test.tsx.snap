// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActuFiltre renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
      "padding": 16,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Ville
  </Text>
  <View
    style={
      [
        {
          "alignSelf": "stretch",
        },
        undefined,
      ]
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="ios_touchable_wrapper"
    >
      <View
        pointerEvents="box-only"
      >
        <TextInput
          editable={false}
          style={
            [
              {
                "backgroundColor": "#fff",
                "borderColor": "#4A6670",
                "borderRadius": 5,
                "borderWidth": 1,
                "height": 40,
                "paddingHorizontal": 10,
              },
              {
                "color": "#4A6670",
              },
            ]
          }
          testID="text_input"
          value="Sélectionnez une Ville"
        />
      </View>
    </View>
  </View>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Date de début
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        [
          {
            "borderColor": "#4A6670",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 10,
          },
          {},
        ]
      }
    >
      Sélectionnez une date
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Date de fin
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        [
          {
            "borderColor": "#4A6670",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 10,
          },
          {},
        ]
      }
    >
      Sélectionnez une date
    </Text>
  </View>
  <Text
    style={
      {
        "fontSize": 18,
        "fontWeight": "bold",
        "marginBottom": 8,
      }
    }
  >
    Plante
  </Text>
  <View
    style={
      {
        "marginBottom": 10,
        "position": "relative",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": false,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          [
            {
              "alignSelf": "stretch",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID="ios_touchable_wrapper"
        >
          <View
            pointerEvents="box-only"
          >
            <TextInput
              editable={false}
              style={
                [
                  {
                    "backgroundColor": "#fff",
                    "borderColor": "#4A6670",
                    "borderRadius": 5,
                    "borderWidth": 1,
                    "height": 40,
                    "paddingHorizontal": 10,
                  },
                  {
                    "color": "#4A6670",
                  },
                ]
              }
              testID="text_input"
              value="Sélectionnez une Plante"
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#668F80",
        "borderRadius": 5,
        "marginTop": 20,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
        }
      }
    >
      Rechercher
    </Text>
  </View>
</View>
`;
